### DISCUSSION TITLE
How do I do dynamic routing for a page at the root like buy-with-[slug].js

### DISCUSSION THREAD

**Discussion by corymsmith** [vercel/next.js#22078] [Help]
I just keep getting a 404 if I try routes like `/buy-with-bitcoin`, etc. so its not even running that pages code.

**Comment by laugharn** [comment]
Next doesn't support parameters in the middle of the route name, per the original RFC #7607

> ### Named parameters cannot appear in the middle of a route name.
> 
> This means a page named blog-[id].js would be interpreted literally and not matched by /blog-1. You can either restructure your page to be /blog/[id].js or turn the entire URL Segment into a named parameter and handle stripping blog- in your application's code.
